### ISSUE TITLE
Break documentation out

### ISSUE THREAD

**Issue by jwerle** [regl-project/regl#381]
The `API.md` file has grown to quite large. What do you guys think about breaking it out into separate files and having an amalgamated build of `API.md` for ease of navigation?

**Comment by mikolalysenko** [comment]
Yeah.  I'm open to suggestions on how best to do this.  I think the next biggest win though is to finish moving all the examples into a separate repo.  This is going to require an extensive audit of all the docs though.

**Comment by jwerle** [comment]
@mikolalysenko I wouldn't mind breaking the examples out a bit. Any thoughts on how it should be done?

**Comment by mikolalysenko** [comment]
The first step is to make sure everything is up to date here and that all the build scripts work:

https://github.com/regl-project/regl-gallery/tree/gh-pages/example

Then we need to change all the links to point there instead of at the main repo and remove all references to the examples from the build scripts.

Finally we should run git-filter-branch and clean up all the old objects from history so the repo size becomes manageable again.

**Comment by jwerle** [comment]
@mikolalysenko I've taken my first stab at breaking the examples out of the main project repository.
Check it out here [littlstar/regl-examples](https://github.com/littlstar/regl-examples).
The source lives in the *master* branch while the built examples live in the `gh-pages` branch for viewing.
The built gallery can be seen here: littlstar.github.io/regl-examples
I've modified the build script in https://github.com/regl-project/regl/pull/384 and have adopted those changes in the new examples build script as well.

**Comment by mikolalysenko** [comment]
Thanks so much for this!

**Comment by gregtatum** [comment]
Docs in one file is kind of nice for grepping for something.
